lib/widgets/nutrition/ingredient_dialogs.dart

```diff
@@ -0,0 +1,199 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:wger/helpers/misc.dart';
+import 'package:wger/models/nutrition/ingredient.dart';
+import 'package:wger/models/nutrition/nutritional_goals.dart';
+import 'package:wger/widgets/nutrition/macro_nutrients_table.dart';
+
+Widget ingredientImage(String url, BuildContext context) {
+  var radius = 100.0;
+  final height = MediaQuery.sizeOf(context).height;
+  final width = MediaQuery.sizeOf(context).width;
+  final smallest = height < width ? height : width;
+  if (smallest < 400) {
+    radius = smallest / 4;
+  }
+  return Padding(
+    padding: const EdgeInsets.only(bottom: 12),
+    child: CircleAvatar(backgroundImage: NetworkImage(url), radius: radius),
+  );
+}
+
+class IngredientDetails extends StatelessWidget {
+  final AsyncSnapshot<Ingredient> snapshot;
+  final void Function()? select;
+  const IngredientDetails(this.snapshot, {super.key, this.select});
+
+  @override
+  Widget build(BuildContext context) {
+    Ingredient? ingredient;
+    NutritionalGoals? goals;
+    String? source;
+
+    if (snapshot.hasData) {
+      ingredient = snapshot.data;
+      goals = ingredient!.nutritionalValues.toGoals();
+      source = ingredient.sourceName ?? 'unknown';
+    }
+
+    return AlertDialog(
+      title: (snapshot.hasData) ? Text(ingredient!.name) : null,
+      content: SingleChildScrollView(
+        child: Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              if (snapshot.hasError)
```

Comment:
I feel we could rewrite this list so that we only have one snapshot.hasData (and .hasError) check and then show all the ingredient and not need to check for hasData on every line, but it works, so 🤷🏻‍♂️